### PR TITLE
[Model] LoRA gptbigcode implementation

### LINKED ISSUE #3011: GPTBigCodeForCausalLM LoRA support
Would love to see LoRA support for GPTBigCodeForCausalLM. We have fine tuned adapters per-language.

### PR DESCRIPTION
**Pull request by raywanb**
Added lora adapters to gptbigcode

Resolves #3011

### CODE REVIEW

File: csrc/punica/bgmv/bgmv_config.h [hunk 1]
```diff
@@ -27,6 +27,7 @@ void bgmv_kernel(out_T *__restrict__ Y, const in_T *__restrict__ X,
     f(in_T, out_T, W_T, narrow, 2752) \
     f(in_T, out_T, W_T, narrow, 2816) \
     f(in_T, out_T, W_T, narrow, 3072) \
+    f(in_T, out_T, W_T, narrow, 3328) \
```
Comment (Yard1) [inline csrc/punica/bgmv/bgmv_config.h:30]: please add the new dimensions to `tests/lora/test_punica.py`

**Review by Yard1 — CHANGES_REQUESTED** [review]
Thanks, please update the tests!

File: csrc/punica/bgmv/bgmv_config.h (hunk 1 above)
Comment (raywanb) [reply csrc/punica/bgmv/bgmv_config.h:30]: just added

**Comment by raywanb** [comment]
> Thanks, please update the tests!

Done!

**Comment by Yard1** [comment]
@raywanb great, all that's left is to run `./format.sh` to make lint pass

**Review by Yard1 — APPROVED** [review]

**Comment by raywanb** [comment]
@Yard1 it's ready to merge!

**Comment by raywanb** [comment]
> @raywanb could you resolve the conflicts one more time 🙏

just did!

**Comment by njhill** [comment]
Thanks @raywanb! Could you please also run the linter on the final changes to fix those errors? (`./format.sh`)

**Comment by njhill** [comment]
@raywanb the lora CI test failures also look suspicious, these aren't currently failing on the main branch. cc @Yard1

**Comment by Yard1** [comment]
Yeah those are real failures...

**Comment by njhill** [comment]
@raywanb thanks for the updates! It looks like the lora tests are still [failing](https://buildkite.com/vllm/ci/builds/7742#018f99f5-0763-4d42-8e3f-a5a69842ea23) though.

There was an issue in the main branch but I don't think that's the cause. Would be good if you could merge in main again once #4894 has been merged and then check the failures..

**Comment by raywanb** [comment]
Yep will look into it later today

**Review by njhill — APPROVED** [review]
Thanks @raywanb! @Yard1 is this still good with you to merge?

**Pull request merged**